src/components/Table/GridTable.tsx

```diff
@@ -1503,7 +1510,7 @@ function filterRows<R extends Kinded>(
       return acc.concat([[row, row.children?.reduce(acceptAll, []) ?? []]]);
     } else {
       const matchedChildren = row.children?.reduce(filterFn, []) ?? [];
-      if (matchedChildren.length > 0 || row.pin) {
+      if (matchedChildren.length > 0 || typeof row.pin === "string" || row.pin?.filter === false) {
```

Comment:
Suggestion: I think `row.pin?.filter !== true` would be slightly better b/c if the user doesn't set `pin.filter`, then we still want to treat it like `filter: false`.